### ISSUE TITLE
"300 degC" wrongly converted to "300 K" by `toSI()`

### ISSUE THREAD

**Issue by jkotan** [josdejong/mathjs#3097]
The  problem is related to #2499 but it was not fixed by the corresponding PR.

When we convert units from `degC` to `K`  in SciCat they are wrongly converted by `toSI()` i.e. without adding the offset (https://github.com/SciCatProject/scicat-backend-next/issues/926 ).

In physics you never use Celsius to describe a difference of two various temperatures by always Kelvin 
so during conversion from `degC` to `K` the offset should be always added.

You can workaround the problem using twice `unit()` , `to()` and `toSi()` but it looks complicated.

Is this behavour expected?

**To Reproduce**
```js
>  const { unit, version } = await import("mathjs");

> version
'12.1.0'

>  unit(300. "degC").toSI().toJSON()
{ mathjs: 'Unit', value: 300, unit: 'K', fixPrefix: true }

>  unit(300, "degC").to(unit("degC").toSI()).toJSON()
{ mathjs: 'Unit', value: 573.15, unit: 'K', fixPrefix: true }
```

**Comment by josdejong** [comment]
Thanks for reporting. At first sight this indeed looks like a bug in `Unit.toSI` to me, this method does not contain any logic to recon with a temperature offset like `Unit.to` does. 

Any thoughts on this @costerwi or @gwhitney?

**Comment by gwhitney** [comment]
I mean the whole situation is super complicated by whether a given units quantity is a specific temperature, or a difference in temperatures. I can't keep straight in my head the convention we were heading to between what units to use for which; I think maybe it was that degC should only be used for specific temperatures, and K for either specific temperature or differences? If so, then yes, using the offset when converting should happen. So sounds like this is a bug. The real best way to handle these things would be to get the modularization with respect to a separate Unit package done, and then put the energy into fixing that Unit package... sorry not to be of more help.

**Comment by josdejong** [comment]
Thanks for your inputs Glen. 

I think the issue here is that the behavior of `toSI()` and `to()` are not consistent:

```js
math.unit("300 degC").toSI().toString()   // "300 K"
math.unit("300 degC").to("K").toString()  // "573.15 K" 
```

Maybe `.toSI()` is just not on par with `.to()`. That could be related to the changes in `.to()` that we made in #2501.

**Comment by josdejong** [comment]
> Maybe it can be rewritten to identify the correct SI unit and then convert to it using .to().

Reusing that logic would be nice indeed.

**Comment by josdejong** [comment]
This issue has been fixed now in `v12.3.0` via #3118.
